lib/doorkeeper/oauth/assertion_access_token_request.rb

```diff
@@ -39,114 +39,119 @@ def authorize
+      #
+      # @see https://tools.ietf.org/html/draft-ietf-oauth-jwt-bearer-12#section-3
+      #
+      def validate_client
+        @client ||= if configuration.jwt_use_issuer_as_client_id
+                      OAuth::Client.find(server.jwt['iss']) if server.jwt['iss'].present?
+                    elsif sever.parameters[:client_id].present?
+                      OAuth::Client.find(sever.parameters[:client_id])
+                    end
+      end
 
-        def validate_resource_owner
-          resource_owner || (@resource_owner = client.try(:application).try(:owner) || server.current_resource_owner)
-        end
+      ##
+      # >   The "scope" parameter may be used, as defined in the Assertion
+      # >   Framework for OAuth 2.0 Client Authentication and Authorization
+      # >   Grants [I-D.ietf-oauth-assertions] specification, to indicate the
+      # >   requested scope.
+      #
+      # @see https://tools.ietf.org/html/draft-ietf-oauth-jwt-bearer-12#section-2.1
+      #
+      def validate_scopes
+        @original_scopes = @original_scopes || server.jwt['scope']
+        return true unless @original_scopes
+        ScopeChecker.valid? @original_scopes, configuration.scopes, client.try(:scopes)
+      end
 
-        def validate_access_token
-          access_token or create_token
-        end
+      def validate_resource_owner
+        resource_owner || (@resource_owner = client.try(:application).try(:owner) || server.current_resource_owner)
+      end
+
+      def validate_access_token
+        access_token or create_token
+      end
 
-        ##
-        # >  [...] refresh tokens are not issued
-        # >  in response to assertion grant requests and access tokens will be
-        # >  issued with a reasonably short lifetime.
-        #
-        # @see https://tools.ietf.org/html/draft-ietf-oauth-assertions-18#section-4.1
-        #
-        def create_token
-          client_requested_expires_in = server.jwt['exp'].to_i - server.jwt['iat'].to_i
-          server_expires_in           = Authorization::Token.access_token_expires_in(configuration, client)
+      ##
+      # >  [...] refresh tokens are not issued
+      # >  in response to assertion grant requests and access tokens will be
+      # >  issued with a reasonably short lifetime.
+      #
+      # @see https://tools.ietf.org/html/draft-ietf-oauth-assertions-18#section-4.1
+      #
+      def create_token
+        client_requested_expires_in = server.jwt['exp'].to_i - server.jwt['iat'].to_i
+        server_expires_in           = Authorization::Token.access_token_expires_in(configuration, client)
+        if server_expires_in
           expires_in = (client_requested_expires_in > 0 && client_requested_expires_in <= server_expires_in) ? client_requested_expires_in : server_expires_in
-          @access_token = AccessToken.find_or_create_for(client, resource_owner.id, scopes, expires_in, configuration.refresh_token_enabled?)
+        else
+          expires_in = nil
```

Comment:
@paulld this is how we make token permanent. 
https://github.com/easyship/easyship-api/pull/2921/files#diff-e621685ac1c2689eaedaedf334c6123bR25